in_toto/runlib.go

```diff
@@ -66,22 +75,88 @@ the following format:
 If recording an artifact fails the first return value is nil and the second
 return value is the error.
 */
-func RecordArtifacts(paths []string) (map[string]interface{}, error) {
+func RecordArtifacts(paths []string) (evalArtifacts map[string]interface{}, err error) {
+	// check if visitedSymlinks has been initialized
+	// if not: generate a new set
+	if visitedSymlinks == nil {
+		visitedSymlinks = NewSet()
+	}
```

Comment:
Do we want to support a use-case where the visitedSymlinks Set is initialized outside of this function? I feel like (re-)initializing it no matter what, or raising an error if it is already initialized come both more expected.

Reply:
We could allow passing the set to the function and if this set is nil we can initialize a global set, but do we still need a global variable and a wrapper if we allow this? And If we allow passing a set to the function, shouldn't the function return a set then?

We could also work with Go receivers (I have no idea if this is supported by our set. It's just a map with empty structs as values.. It seems to be possible according to: https://stackoverflow.com/questions/49176090/map-as-a-method-receiver):
```
func (visitedSymlinks Set) RecordArtifacts(...)
```

Reply:
Ah ok.. Now I understood your second sentence, sorry. Yes it makes sense to always initialize it no matter what. Checking for nil, is wrong here, because we want to make sure that the set is always fresh initialized. I think this is an artifact from renaming the methods.. this would have made sense if we call RecordArtifacts (the wrapper) recursively, but we don't.

Reply:
I am going to fix this